src/cljs/rems/user_settings.cljs

```diff
@@ -80,10 +80,22 @@
                    (when first-time? ; to avoid an infinite loop if the settings fail to save (e.g. unsupported language in cookies)
                      [[:check-if-should-save-language!]]))})))
 
+(rf/reg-event-fx
+ :loaded-user-settings-fail
+ (fn [{:keys [db]} [_ user-settings-error]]
+   (let [logged-in? (:user (:identity db))]
+     (if-not logged-in?
+       (if-not (= 401 (:status user-settings-error))
+         {:db db
+          :dispatch (flash-message/show-default-error! :top (str "Fetch user settings"))}
+         {:db db})
+       {:db db}))))
+
 (defn fetch-user-settings! []
   (fetch "/api/user-settings"
-         {:handler #(rf/dispatch-sync [:loaded-user-settings %])
-          :error-handler (flash-message/default-error-handler :top "Fetch user settings")}))
+         {:simple-error-handler? true
+          :handler #(rf/dispatch-sync [:loaded-user-settings %])
+          :error-handler #(rf/dispatch [:loaded-user-settings-fail %])}))
```

Comment:
Can just inline the code here.

Reply:
I would leave it like this, if you do not have a strong opinion on it.

Reply:
I think it looks rather consistent this way.

Reply:
Yeah that is right, the simplification is elsewhere.